test/ArcxAnalyticsProvider.test.tsx

```diff
@@ -177,7 +173,10 @@ describe('(int) ArcxAnalyticxProvider', () => {
 
         expect(socketStub.emit).calledOnceWith(
           'submit-event',
-          getClickEventBody('div#id-for-click.test-classname-1.test-classname-2', 'Text to click'),
+          getClickEventBody(
+            'div#id-for-click.test-classname-1.test-classname-2 [data-testid=track-click] [class=test-classname-1 test-classname-2]',
```

Comment:
- Tag `id`s can contain a dot in them, so this format makes it confusing on where the tag ID stops. We should add a delimiter between the id and the list of classes, such as `;`.
- Do not add spaces between attributes `[..]`
- There's no need to add the `class` attribute

As a result of these points, this is what we should expect here:
`div#id-for-click;.test-classname-1.test-classname-2[data-testid=track-click]`

Reply:
To give you an idea, Heap does it this way, as an example:

```
@div;#modal-root;|@div;.kbmAfw;.sc-hTBuwn;|@div;.jLzChf;.sc-jfTVlA;|@div;|@div;|@div;.bOhFXr;.sc-iOeugr;|@input;.nfYIP;.sc-lbVpMG;[placeholder=Amount];[type=text];|
```

So it separates each tag by a `|`, prefixes each tag with `@`, each id with `#` and separates each item by a `;`. We don't have to necessarily copy them. If you can think of a better way to do it we can go for it

Reply:
so maybe we can just move id tag to attributes, will it be clearer?
According to dot, I got this approach from browser identifier:
<img width="529" alt="image" src="https://user-images.githubusercontent.com/18648454/219339626-4876f508-3a00-4515-bf82-907175d01123.png">
I will remove spaces between attributes